Gradle requires JAVA_HOME to be set

I get the following error when running the build:

```
* What went wrong:
A problem occurred configuring project ':distribution:tar'.
> Ambiguous method overloading for method java.io.File#<init>.
  Cannot resolve which method to invoke for [null, class java.lang.String] due to overlapping prototypes between:
  	[class java.lang.String, class java.lang.String]
  	[class java.io.File, class java.lang.String]
```

which is due to the following problem:

```
Caused by: groovy.lang.GroovyRuntimeException: Ambiguous method overloading for method java.io.File#<init>.
Cannot resolve which method to invoke for [null, class java.lang.String] due to overlapping prototypes between:
	[class java.lang.String, class java.lang.String]
	[class java.io.File, class java.lang.String]
	at org.elasticsearch.gradle.test.ClusterFormationTasks$_configureCheckPreviousTask_closure16.doCall(ClusterFormationTasks.groovy:282)
	at org.gradle.api.internal.ClosureBackedAction.execute(ClosureBackedAction.java:67)
	at org.gradle.util.ConfigureUtil.configure(ConfigureUtil.java:130)
	at org.gradle.util.ConfigureUtil.configure(ConfigureUtil.java:110)
	at org.gradle.api.internal.AbstractTask.configure(AbstractTask.java:488)
	at org.gradle.api.internal.tasks.DefaultTaskContainer.create(DefaultTaskContainer.java:93)
	at org.gradle.api.tasks.TaskContainer$create$3.call(Unknown Source)
	at org.elasticsearch.gradle.test.ClusterFormationTasks.configureCheckPreviousTask(ClusterFormationTasks.groovy:278)
	at org.elasticsearch.gradle.test.ClusterFormationTasks.configureTasks(ClusterFormationTasks.groovy:94)
	at org.elasticsearch.gradle.test.ClusterFormationTasks$configureTasks$1.callStatic(Unknown Source)
	at org.elasticsearch.gradle.test.ClusterFormationTasks.setup(ClusterFormationTasks.groovy:49)
	at org.elasticsearch.gradle.test.ClusterFormationTasks$setup.call(Unknown Source)
	at org.elasticsearch.gradle.test.RestIntegTestTask$_closure1.doCall(RestIntegTestTask.groovy:74)
	at org.gradle.listener.ClosureBackedMethodInvocationDispatch.dispatch(ClosureBackedMethodInvocationDispatch.java:40)
	at org.gradle.listener.ClosureBackedMethodInvocationDispatch.dispatch(ClosureBackedMethodInvocationDispatch.java:25)
	at org.gradle.internal.event.AbstractBroadcastDispatch.dispatch(AbstractBroadcastDispatch.java:44)
	at org.gradle.internal.event.BroadcastDispatch.dispatch(BroadcastDispatch.java:79)
	at org.gradle.internal.event.BroadcastDispatch.dispatch(BroadcastDispatch.java:30)
	at org.gradle.messaging.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:93)
	at com.sun.proxy.$Proxy12.afterEvaluate(Unknown Source)
	at org.gradle.configuration.project.LifecycleProjectEvaluator.notifyAfterEvaluate(LifecycleProjectEvaluator.java:67)
	... 41 more
```

And indeed at ClusterFormationTasks.groovy:282 we have the following

```groovy
            commandLine new File(System.getenv('JAVA_HOME'), 'bin/jps'), '-l'
```

So if JAVA_HOME is not set then Groovy won't know whether to use the File(File, String) or the File(String, String) constructor. Could we just expect jps to be on the PATH or at least detect JAVA_HOME automatically when it is not set?

--- a/buildSrc/src/main/groovy/org/elasticsearch/gradle/test/ClusterFormationTasks.groovy
+++ b/buildSrc/src/main/groovy/org/elasticsearch/gradle/test/ClusterFormationTasks.groovy
@@ -279,7 +279,7 @@ class ClusterFormationTasks {
             onlyIf { pidFile.exists() }
             // the pid file won't actually be read until execution time, since the read is wrapped within an inner closure of the GString
             ext.pid = "${ -> pidFile.getText('UTF-8').trim()}"
-            commandLine new File(System.getenv('JAVA_HOME'), 'bin/jps'), '-l'
+            commandLine new File(System.getProperty('java.home'), 'bin/jps'), '-l'
             standardOutput = new ByteArrayOutputStream()
             doLast {
                 String out = standardOutput.toString()

